src/coreclr/jit/codegenarm64.cpp

```diff
@@ -4680,6 +4680,57 @@ void CodeGen::genCodeForSelect(GenTreeOp* tree)
     genProduceReg(tree);
 }
 
+//------------------------------------------------------------------------
+// genCodeForCinc: Produce code for a GT_CINC/GT_CINCCC node.
+//
+// Arguments:
+//    tree - the node
+//
+void CodeGen::genCodeForCinc(GenTreeOp* cinc)
+{
+    assert(cinc->OperIs(GT_CINC, GT_CINCCC));
+
+    GenTree* opcond = nullptr;
+    GenTree* op     = cinc->gtOp1;
+    if (cinc->OperIs(GT_CINC))
+    {
+        opcond = cinc->gtOp1;
+        op     = cinc->gtOp2;
+        genConsumeRegs(opcond);
+    }
+
```

Comment:
We seem to be missing a call to `genConsumeRegs(op)`

Reply:
Aren't we consuming regs [here](https://github.com/dotnet/runtime/pull/82031/files#diff-684b0e9641db2a0bc52824ae27e3c9f81dd4e6738441a8258c2d4cc1f2fa18eaR4726) correctly?

Reply:
No, it should be done before you start emitting code. Consuming the operands can emit code that should happen before the codegen for this node begins.

Also, you should mark the 0 as contained in lowering to avoid taking up a register unnecessarily. I'd suggest something like the following:
```cpp
genConsumeRegs(op);

if (op->isContained())
{
  assert(op->IsIntegralConst(0));
  srcReg = REG_ZR;
}
else
{
  srcReg = op->GetRegNum();
}
```

Reply:
> Also, you should mark the 0 as contained in lowering to avoid taking up a register unnecessarily.

Ah, that part is probably unnecessary if the containment is just inherited from previous `SELECT` handling.

Reply:
Done